src/addons/filter/mediaplugin/classes/videojs-ogvjs.ts

```diff
@@ -0,0 +1,695 @@
+// (C) Copyright 2015 Moodle Pty Ltd.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import { CoreApp } from '@services/app';
+import { CoreObject } from '@singletons/object';
+import { OGVPlayer, OGVCompat, OGVLoader } from 'ogv';
+import videojs, { PreloadOption, TechSourceObject, VideoJSOptions } from 'video.js';
+
+export const Tech = videojs.getComponent('Tech');
+
+/**
+ * OgvJS Media Controller for VideoJS - Wrapper for ogv.js Media API.
+ *
+ * Code adapted from https://github.com/HuongNV13/videojs-ogvjs.
+ */
+export class VideoJSOgvJS extends Tech {
+
+    /**
+     * List of available events of the media player.
+     */
+    // eslint-disable-next-line @typescript-eslint/naming-convention
+    static readonly Events = [
+        'loadstart',
+        'suspend',
+        'abort',
+        'error',
+        'emptied',
+        'stalled',
+        'loadedmetadata',
+        'loadeddata',
+        'canplay',
+        'canplaythrough',
+        'playing',
+        'waiting',
+        'seeking',
+        'seeked',
+        'ended',
+        'durationchange',
+        'timeupdate',
+        'progress',
+        'play',
+        'pause',
+        'ratechange',
+        'resize',
+        'volumechange',
+    ];
+
+    protected playerId: string;
+    protected parentElement: HTMLElement | null = null;
+    protected placeholderElement = document.createElement('div');
+
+    // Variables/functions defined in parent classes.
+    protected el_!: OGVPlayerEl; // eslint-disable-line @typescript-eslint/naming-convention
+    protected options_!: VideoJSOptions; // eslint-disable-line @typescript-eslint/naming-convention
+    protected currentSource_?: TechSourceObject; // eslint-disable-line @typescript-eslint/naming-convention
+    protected triggerReady!: () => void;
+    protected on!: (name: string, callback: (e?: Event) => void) => void;
+
+    /**
+     * Create an instance of this Tech.
+     *
+     * @param options The key/value store of player options.
+     * @param ready Callback function to call when the `OgvJS` Tech is ready.
+     */
+    constructor(options: VideoJSOptions & { playerId: string }, ready: () => void) {
```

Comment:
```suggestion
    constructor(options: VideoJSTechOptions, ready: () => void) {
```

I think it's better to extract this into a different type, even if it's only used once and not exported from this file. When looking at the code, I was confused where this `playerId` comes from and it took me a while to realize that this is actually [called internally by VideoJS](https://github.com/videojs/video.js/blob/b306ce614e70e6d3305348d1b69e1434031d73ef/src/js/player.js#L1194).